tests/prague/eip6110_deposits/test_modified_contract.py

```diff
@@ -0,0 +1,201 @@
+    # "transactionHash":"0x2d71f3085a796a0539c9cc28acd9073a67cf862260a41475f000dd101279f94f",
+    # "blockHash":"0x8062a17fa791f5dbd59ea68891422e3299ca4e80885a89acf3fc706c8bceef53",
+    # "blockNumber":"0x794fb5",
+    # "address":"0x7f02c3e3c98b133055b8b348b2ac625669ed295d",
+    # "data":"0x0000000000000000000000000000000000000000000000000000000000000001",
+    # "topics":["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
+    # "0x0000000000000000000000006885e36bfcb68cb383dfe90023a462c03bcb2ae5",
+    # "0x00000000000000000000000080b5dc88c98e528bf9cb4b7f0f076ac41da24651"]
+
+    bytecode = Op.LOG3(
+        # ERC-20 token transfer log
+        # ERC-20 token transfers are LOG3, since the topic, the sender, and receiver
+        # are all topics (the sender and receiver are `indexed` in the solidity event)
+        0,
+        32,
+        0xDDF252AD1BE2C89B69C2B068FC378DAA952BA7F163C4A11628F55A4DF523B3EF,
+        0x000000000000000000000000AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA,
+        0x000000000000000000000000BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB,
+    )
+
+    requests = Requests()
+
+    if include_deposit_event:
+        bytecode += Om.MSTORE(deposit_request_log) + Op.LOG1(
+            0,
+            len(deposit_request_log),
+            Spec.DEPOSIT_EVENT_SIGNATURE_HASH,
+        )
+        requests = Requests(deposit_request)
+    bytecode += Op.STOP
+
+    pre[Spec.DEPOSIT_CONTRACT_ADDRESS] = Account(
+        code=bytecode,
+        nonce=1,
+        balance=0,
+    )
+    sender = pre.fund_eoa()
+
+    tx = Transaction(
+        to=Spec.DEPOSIT_CONTRACT_ADDRESS,
+        sender=sender,
+        gas_limit=100_000,
+    )
+
+    blockchain_test(
+        pre=pre,
+        blocks=[
+            Block(
+                txs=[tx],
+                header_verify=Header(
+                    requests_hash=requests,
+                ),
+            ),
+        ],
+        post={},
+    )
+
+
+@pytest.mark.parametrize(
+    "log_argument,value",
+    [(log_argument, value) for log_argument in EVENT_ARGUMENTS for value in EVENT_ARGUMENT_VALUES],
+)
+def test_invalid_layout(
+    blockchain_test: BlockchainTestFiller, pre: Alloc, log_argument: str, value: str
+):
+    """Test deposit contract emitting logs with invalid layouts (sizes/offsets)"""
+
+    log_params = {**DEFAULT_DEPOSIT_REQUEST_LOG_DATA_DICT}
+    log_params[log_argument] = 0 if value == "zero" else int("FF" * 32, 16)
+
+    deposit_request_log = create_deposit_log_bytes(**log_params)
+
+    bytecode = Om.MSTORE(deposit_request_log) + Op.LOG1(
+        0,
+        len(deposit_request_log),
+        Spec.DEPOSIT_EVENT_SIGNATURE_HASH,
+    )
+    bytecode += Op.STOP
+
+    pre[Spec.DEPOSIT_CONTRACT_ADDRESS] = Account(
+        code=bytecode,
+        nonce=1,
+        balance=0,
+    )
+    sender = pre.fund_eoa()
+
+    tx = Transaction(
+        to=Spec.DEPOSIT_CONTRACT_ADDRESS,
+        sender=sender,
+        gas_limit=100_000,
```

Comment:
```suggestion
        gas_limit=100_000,
        error=BlockException.INVALID_REQUESTS,
```
We have to add the exception here too due to how EELS applies the logic verification to the deposit contract (has to be the same as the exception at the block level so we could end up changing it).